server/src/routes/api/v1/votesRouter.js

```diff
@@ -0,0 +1,37 @@
+import express from "express"
+import { Vote, Review } from "./../../../models/index.js"
+
+
+const votesRouter = new express.Router()
+votesRouter.post("/", async (req, res) => {
+  const { currentReviewId,  voteDirection } = req.body
+  let currentReviewIdInteger = parseInt(currentReviewId)
+  let currentUserIdInteger = parseInt(req.user.id)
+  try {
+    const newVote = await Vote.query().findOne({ userId: currentUserIdInteger, reviewId: currentReviewId  })
+   
+    if (!newVote) {
+       await Vote.query().insert({
+        userId: currentUserIdInteger,
+        reviewId: currentReviewIdInteger,
+        voteValue: voteDirection
+      })
+    } else {
+      if (newVote.voteValue === voteDirection) {
+         await Vote.query().updateAndFetchById(newVote.id, { voteValue: 0 }) 
+      } else {
+        await Vote.query().updateAndFetchById(newVote.id, { voteValue: voteDirection }) 
+      } 
+    }
+
+    const review = await Review.query().findById(currentReviewIdInteger)
+    
+    const voteTotals = await review.$voteCount()
+    
+    res.status(200).json({ currentUserVoteValue: voteTotals })
```

Comment:
that property `currentUserVoteValue` doesnt make sense in this context. I would literally call it `voteTotals` or better yet `VoteCount`, which is what you have when you initially retrieve your vote information on page load